src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -181,32 +181,22 @@ HRESULT STDMETHODCALLTYPE CorProfiler::ModuleLoadFinished(ModuleID module_id,
 
   // Identify the AppDomain ID of mscorlib which will be the Shared Domain
   // because mscorlib is always a domain-neutral assembly
-  if (!corlib_module_loaded && (module_info.assembly.name == "mscorlib"_W ||
+  if (!corlib_module_loaded &&
+      (module_info.assembly.name == "mscorlib"_W ||
        module_info.assembly.name == "System.Private.CoreLib"_W)) {
     corlib_module_loaded = true;
     corlib_app_domain_id = app_domain_id;
-    corlib_module_id = module_id;
     return S_OK;
   }
 
   // Identify the AppDomain ID of the managed profiler entrypoint
   if (module_info.assembly.name == "Datadog.Trace.ClrProfiler.Managed"_W) {
-    managed_profiler_module_loaded = true;
-    managed_profiler_app_domain_id = app_domain_id;
-    managed_profiler_module_id = module_id;
-  }
+    managed_profiler_loaded_app_domains.insert(app_domain_id);
 
-  // Do not modify the module if it has been loaded into the Shared Domain
-  // and the profiler is not in the Shared Domain
-  if (runtime_information_.is_desktop() &&
-      corlib_module_loaded && managed_profiler_module_loaded &&
-      app_domain_id == corlib_app_domain_id &&
-      corlib_app_domain_id != managed_profiler_app_domain_id) {
-    Info(
-        "ModuleLoadFinished skipping modifying assembly because it is "
-        "domain-neutral but the managed profiler is not: ",
-        module_id, " ", module_info.assembly.name);
-    return S_OK;
+    if (runtime_information_.is_desktop() && corlib_module_loaded &&
+        app_domain_id == corlib_app_domain_id) {
```

Comment:
Do you think this could open us up to crash conditions?

Reply:
I think you're commenting on the fact that we do not exit early here, is that correct? We already changed this method so that we no longer add any AssemblyReferences in this method, so the result of this change is that we will now save `ModuleMetadata` for these modules and we will consider them for instrumentation during `JITCompilationStarted`. Before, when we returned we did not save the `ModuleMetadata` so in `JITCompilationStarted` we exited early without fully considering methods inside this module.

To prevent crashes, I've introduced changes in `JITCompilationStarted` so we should not have regressions.